runtime/jcl/common/jcldefine.c

```diff
@@ -245,3 +245,179 @@ defineClassCommon(JNIEnv *env, jobject classLoaderObject,
 	return NULL;
 #endif /* J9VM_OPT_DYNAMIC_LOAD_SUPPORT */
 }
+
+jclass
+defineClassAnonymous(JNIEnv *env, jobject classLoaderObject, jbyteArray classRep, jint length, jobject protectionDomain, UDATA options, J9Class *hostClass, jobjectArray cpPatch)
```

Comment:
Why duplicate `defineClassCommon()`?  Duplicating it will require keeping both implementations in sync forever.  Instead, can you add a `patchRAMConstantPool(J9Class *clazz, jobjectArray cpPatch)` method that handles the ramCP patching?